src/main/haskell/kore/src/Kore/Step/Function/Identifier.hs

```diff
@@ -0,0 +1,47 @@
+{-|
+Module      : Kore.Step.Function.Identifier
+Description : Data structures used for axiom evaluation.
+Copyright   : (c) Runtime Verification, 2019
+License     : NCSA
+Maintainer  : virgil.serbanuta@runtimeverification.com
+Stability   : experimental
+Portability : portable
+
+TODO(virgil): Move from Kore.Step.Function to Kore.Step.Axiom
+-}
+
+module Kore.Step.Function.Identifier
+    ( AxiomIdentifier (..)
+    , extract
+    ) where
+
+import Kore.AST.Common
+       ( SymbolOrAlias (..) )
+import Kore.AST.Identifier
+       ( Id (..) )
+import Kore.AST.Pure
+       ( PurePattern )
+import Kore.AST.Valid
+       ( pattern App_, pattern Ceil_ )
+
+data AxiomIdentifier level
+    = Application !(Id level)
```

Comment:
Could we call these constructors `ApplicationId` and `CeilId` to avoid collision with the `Application` and `Ceil` constructors?